core/src/main/java/hudson/slaves/SlaveComputer.java

```diff
@@ -549,10 +543,20 @@ public void onClosed(Channel c, IOException cause) {
         String slaveVersion = channel.call(new SlaveVersion());
         log.println("Remoting version: " + slaveVersion);
         VersionNumber agentVersion = new VersionNumber(slaveVersion);
+
+        boolean rejectConn = GlobalConfiguration.all().get(SlaveComputerConfiguration.class).isRejectConn();
+
         if (agentVersion.isOlderThan(RemotingVersionInfo.getMinimumSupportedVersion())) {
-            log.println(String.format("WARNING: Remoting version is older than a minimum required one (%s). " +
-                    "Connection will not be rejected, but the compatibility is NOT guaranteed",
-                    RemotingVersionInfo.getMinimumSupportedVersion()));
+            if(!rejectConn)
+                log.println(String.format("Remoting version is older than a minimum required one (%s) " +
```

Comment:
Why have you rewritten the message here?

Reply:
To let the user know that remoting version is unsupported and Jenkins is still trying to connect. If there will be any connection problem them user should know why that happened.